src/sentry/integrations/slack/notifications.py

```diff
@@ -31,28 +31,46 @@ def get_context(
 
 def get_integrations_by_recipient_id(
     organization: Organization, recipients: AbstractSet[Union[User, Team]]
-) -> Mapping[Union[User, Team], ExternalActor]:
-    actor_mapping = {recipient.actor_id: recipient for recipient in recipients}
+) -> Mapping[Union[User, Team], Union[ExternalActor, Identity]]:
 
-    external_actors = ExternalActor.objects.filter(
-        provider=ExternalProviders.SLACK.value,
-        actor_id__in=[recipient.actor_id for recipient in recipients],
-        organization=organization,
-    ).select_related("integration")
-
-    return {
-        actor_mapping.get(external_actor.actor_id): external_actor
-        for external_actor in external_actors
-    }
+    output = {}
+    for recipient in recipients:
+        if isinstance(recipient, User):
+            try:
+                identity = Identity.objects.get(
+                    idp__type=EXTERNAL_PROVIDERS[ExternalProviders.SLACK],
+                    user=recipient.id,
+                )
+            except Identity.DoesNotExist:
+                # the user may not have linked their identity so just move on
+                # since there are likely other users or teams in the list of recipients
+                continue
+            output[recipient] = identity
+        else:
+            external_actor = ExternalActor.objects.filter(
+                provider=ExternalProviders.SLACK.value,
+                actor_id=recipient.actor_id,
+                organization=organization,
+            ).select_related("integration")[0]
+            output[recipient] = external_actor
+    return output
 
 
 def get_channel_and_token(
-    external_actors_by_recipient: Mapping[Union[User, Team], ExternalActor],
+    external_identity_by_recipient: Mapping[Union[User, Team], Union[ExternalActor, Identity]],
```

Comment:
was trying to come up with a combo of `external_actor` and `identity`, open to suggestions